tools/bcdb_mock/pkg/directory/farms_store.go

```diff
@@ -14,14 +17,31 @@ import (
 // FarmAPI holds farm releated handlers
 type FarmAPI struct{}
 
+type farmQuery struct {
+	FarmName string
+	OwnerID  int64
+}
+
+func (f *farmQuery) Parse(r *http.Request) mw.Response {
+	var err error
+	f.OwnerID, err = models.QueryInt(r, "owner")
+	if err != nil {
+		return mw.BadRequest(errors.Wrap(err, "owner should be a integer"))
+	}
+	f.FarmName = r.FormValue("name")
+	return nil
+}
+
 // List farms
 // TODO: add paging arguments
-func (s *FarmAPI) List(ctx context.Context, db *mongo.Database, tid int64, opts ...*options.FindOptions) ([]directory.Farm, int64, error) {
+func (s *FarmAPI) List(ctx context.Context, db *mongo.Database, q farmQuery, opts ...*options.FindOptions) ([]directory.Farm, int64, error) {
```

Comment:
Please change `farmQuery` to `FarmQuery` so the directory package can be used independently from other tools. Now it's a public method that uses a private type as an argument